prboom2/src/m_menu.c

```diff
@@ -5000,7 +5000,10 @@ dboolean M_Responder (event_t* ev) {
     }
     else if (ch > 0)
     {
-      ch = toupper(ch);
+      if (shiftdown && ch >= 32 && ch <= 127)
```

Comment:
I suppose you could move this inside the `if` below since the range check is done by that point.